marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -683,6 +705,12 @@ public Boolean call() {
 
             initEvent(output, callTime);
             batcher.sendSuccessToListeners(output);
+            if(batcher.inputType == Void.class) {
+                CallArgs newInput = batcher.callArgsGenerator.apply(output);
```

Comment:
We need to consider how to respond to errors in the user code in the call generator and, for that matter, the listeners.

Please add a TODO to check how WriteBatcher and QueryBatcher tolerate listener errors.